core/src/main/java/io/aiven/klaw/validation/TopicRequestValidatorImpl.java

```diff
@@ -1,21 +1,244 @@
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  private boolean checkIfPromotionOfTopic(
+      List<Topic> topics,
+      int tenantId,
+      TopicRequestModel topicRequestModel,
+      String syncCluster,
+      ConstraintValidatorContext constraintValidatorContext) {
+    String orderOfEnvs = mailService.getEnvProperty(tenantId, "ORDER_OF_ENVS");
+    boolean promotionOrderCheck = false;
+    if (null != orderOfEnvs) {
+      promotionOrderCheck = checkInPromotionOrder(topicRequestModel.getEnvironment(), orderOfEnvs);
+    }
+
+    if (topics != null && topics.size() > 0) {
+      if (promotionOrderCheck) {
+        int devTopicFound =
+            (int)
+                topics.stream()
+                    .filter(topic -> Objects.equals(topic.getEnvironment(), syncCluster))
+                    .count();
+        if (devTopicFound != 1) {
+          if (topicControllerService.getEnvDetails(syncCluster) == null) {
+            updateConstraint(
+                constraintValidatorContext, "Failure. Base cluster is not configured.");
+          } else {
+            updateConstraint(
+                constraintValidatorContext,
+                "Failure. This topic does not exist in "
+                    + topicControllerService.getEnvDetails(syncCluster).getName()
+                    + " cluster.");
+          }
+          return false;
+        }
+      }
+    } else if (!Objects.equals(topicRequestModel.getEnvironment(), syncCluster)) {
+      if (promotionOrderCheck) {
+        updateConstraint(
+            constraintValidatorContext,
+            "Failure. Please request for a topic first in "
+                + topicControllerService.getEnvDetails(syncCluster).getName()
+                + " cluster.");
+        return false;
+      }
+    }
+
     return true;
-    // TODO move logic from service layer to here for validating topic request
+  }
+
+  private boolean validateTopicConfigParameters(
+      TopicRequestModel topicRequestReq, ConstraintValidatorContext constraintValidatorContext) {
+    log.debug("Into validateTopicConfigParameters");
+
+    String topicPrefix = null, topicSuffix = null;
+    String otherParams =
+        topicControllerService.getEnvDetails(topicRequestReq.getEnvironment()).getOtherParams();
+    String[] params;
+    try {
+      if (otherParams != null) {
+        params = otherParams.split(",");
+        for (String param : params) {
+          if (param.startsWith("topic.prefix")) {
+            topicPrefix = param.substring(param.indexOf("=") + 1);
+          } else if (param.startsWith("topic.suffix")) {
+            topicSuffix = param.substring(param.indexOf("=") + 1);
+          }
+        }
+      }
+    } catch (Exception e) {
+      log.error("Unable to set topic partitions, setting default from properties.", e);
+    }
+
+    try {
+      if (topicPrefix != null
+          && topicPrefix.length() > 0
```

Comment:
Same question about allowing to have prefix consisting of spaces or tabs...
In case it is not allowed it's better to use
```suggestion
          && !topicPrefix.isBlank()
```